### PR TITLE
feat(server): add base server with health checks

### PR DESCRIPTION
**Pull request by shraddha1704**
### What
Add base server with health checks

### Why
Goyagi training

### CODE REVIEW

File: "net/http.lock [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
```
Comment (kylekwong) [inline "net/http.lock]: I'm not sure why this file is called `"net/http.lock` and not `Gopkg.lock`. Can you delete this file and run `dep ensure` again?

File: cmd/serve/main.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+	log.Info("server started")
+
+	err := srv.ListenAndServe()
+	if err != nil && err != http.ErrServerClosed {
+		log.Err(err).Fatal("server stopped")
+	}
+
+	log.Info("server stopped")
+}
```
Comment (kylekwong) [inline cmd/serve/main.go]: Can you make sure all your files end with a newline character? In VSCode, you can set the `    "files.insertFinalNewline"` setting to `true` to do this automatically.

File: pkg/signals/signals.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+// Setup registers for SIGTERM and SIGINT. A channel is returned
+// which is closed on one of these signals. If a second signal is caught,
+// the program is terminated with exit code 1.
+func Setup() <-chan struct{} {
+	close(onlyOneSignalHandler) //panics when called twice
+
+	stop := make(chan struct{}) //create the channel we will return to the callee
+
+	c := make(chan os.Signal, 2)											// create channel c that will receive the os.Signals
```
Comment (kylekwong) [inline pkg/signals/signals.go]: Personally I don't like inline comments at all, but if you want to keep these, I'd say only have one space between your code and the start of the comment, e.g.

```golang
c := make(chan os.Signal, 2) // create channel c that will receive the os.Signals
```

File: pkg/signals/signals.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+// trapSignals(). If trapSignals() is ever invoked again the application will
+// panic trying to close a channel that is already closed. This is to ensure
+// we're only trapping the singals in one place.
+var onlyOneSignalHandler = make(chan struct{})
+
+// Setup registers for SIGTERM and SIGINT. A channel is returned
+// which is closed on one of these signals. If a second signal is caught,
+// the program is terminated with exit code 1.
+func Setup() <-chan struct{} {
+	close(onlyOneSignalHandler) //panics when called twice
```
Comment (kylekwong) [inline pkg/signals/signals.go]: Personal opinion, but I always like having a space between the `//` and the comment, e.g. `// panics when called twice`

**Review by kylekwong — CHANGES_REQUESTED** [review]

File: "net/http.lock (hunk 1 above)
Comment (shraddha1704) [reply "net/http.lock]: yes. I have been thinking about how this too!

File: cmd/serve/main.go (hunk 2 above)
Comment (shraddha1704) [reply cmd/serve/main.go]: Yes. I have added that in the settings. Will be updating them soon!

**Review by kylekwong — APPROVED** [review]

**Pull request merged**